### PR TITLE
[MRG+1] Coverage config

### PR DESCRIPTION
**Pull request by dopplershift**
- Cleaning up codecov config to remove commenting
- Remove coveralls
- Trying to set up codecov to report coverage in test code separately
- Also enabled coverage on AppVeyor

### CODE REVIEW

**Comment by dopplershift** [comment]
If this works first time through I'll be amazed...

**Comment by tacaswell** [comment]
Not liking something about the configuration on windows.

```
    config_read = config.from_file(fname, section_prefix=prefix)
  File "C:\Miniconda35-x64\envs\test-environment\lib\site-packages\coverage\config.py", line 248, in from_file
    section, unknown, filename

```

**Comment by QuLogic** [comment]
Adding tests to coverage fixes #6902.

**Comment by QuLogic** [comment]
So if I understand this correctly; codecov should pass if tests are 100% covered. How is this true, if we aren't technically fully running on Python 2.7 at the moment?

**Comment by dopplershift** [comment]
It's not working properly yet. There should be 3 separate statuses:
- Overall library coverage
- % of the diff covered
- Lines covered in the tests

Not sure why it didn't work, since this was verbatim from my own project with just some paths changed...

**Comment by dopplershift** [comment]
The way this works is that codecov merges coverage runs...so if every line is run at least once, you'll get 100% coverage. We won't be able to see if we're not running things on py2.

It looks like there is a feature called flags where we could mark each run as python 2 or 3. We *could* then segment that way, but then if we have version-specific test code, we would never get to 100%...

**Comment by QuLogic** [comment]
If it's anything like the PR to stop codecov from commenting, we might need to actually merge it to see any effect.

**Comment by QuLogic** [comment]
Oh wait, maybe it's because we've already got a `ci/codecov.yml`?

**Comment by dopplershift** [comment]
Yeah, the duplication is probably the problem. Let's see how this one goes.

**Comment by dopplershift** [comment]
Still didn't take. I've rebased on master (squashed some commits) and fixed a couple minor validation problems with the yaml file I made. If this doesn't seem to take, my only thoughts are to try merging with master and hoping for the best.

**Comment by dopplershift** [comment]
Bingo! And somehow we only have 97% coverage of test files...

Should we commit (maybe with a different target) and try to fix it elsewhere?

**Comment by tacaswell** [comment]
Can we push the test coverage limit down before merging?

**Comment by NelleV** [comment]
We've got 3% of our tests not running on travis? That's interesting to know.

**Comment by QuLogic** [comment]
It's mostly backends (e.g., Qt4/Qt5 can't be run in same process or installed in the same conda environment), some pickle and animation stuff, and everything else seems small.

But when looking at the file on codecov, it says 'File not found in report', so I can't see what the actual problematic lines are. @dopplershift any ideas why it's not coming up properly?

**Comment by dopplershift** [comment]
@NelleV Well, 2.2% of the lines of code in our tests are not being executed anyways.

@QuLogic I think it's failing when it tries to compare against previous coverage on the matplotlib repo, which was excluding the tests.

I've rebased to address the conflicts in `.travis.yml` and `appveyor.yml`. I also dropped the limit for the tests to 97.7%. I'm almost more interested in the files that are a few tenths below 100% since I *really* don't understand why in those cases.

**Comment by dopplershift** [comment]
Rebased now that #7974 is merged. Hopefully everything will work as expected.

**Comment by dopplershift** [comment]
I think we're good now. I did notice at least one coverage change caused by the fact that we have a wait loop (for locks in `cbook.py`) that changes our coverage depending on whether we actually wait.

**Comment by QuLogic** [comment]
All AppVeyor builds show "No coverage report found".

**Comment by dopplershift** [comment]
Was submitting the coverage *after* we cleaned the repos for building packages--which nuked the coverage results. I've now added the upload to the testing phase. (I'm open to other suggestions for fixing this.)

Maybe now everything will work.

**Comment by dopplershift** [comment]
Ok, it looks to be properly running on AppVeyor now--though only for pytest=True. I thought we were entirely using pytest now? (Irrelevant for this PR either way.)

File: appveyor.yml [hunk 1]
```diff
@@ -140,6 +140,8 @@ test_script:
   - if x%USE_PYTEST% == xno python tests.py %PYTEST_ARGS%
   # Generate a html for visual tests
   - python visual_tests.py
+  - if x%USE_PYTEST% == xyes pip install codecov
+  - if x%USE_PYTEST% == xyes codecov -e PYTHON_VERSION PLATFORM
```
Comment (dstansby) [inline appveyor.yml]: These lines mean that codecov only happens if `USE_PYTEST` is true, which is only for the one build. pytest is still being run in the other builds, just through `tests.py` in the root which calls `test()` in `matplotlib/__init__.py`.

Comment (QuLogic) [inline appveyor.yml]: Yea, these conditions could be dropped.

**Pull request merged**